Source/WebCore/svg/SVGPathByteStream.h

```diff
@@ -39,71 +40,158 @@ namespace WebCore {
 class SVGPathByteStream : public CanMakeSingleThreadWeakPtr<SVGPathByteStream> {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    typedef Vector<uint8_t> Data;
-    typedef Data::const_iterator DataIterator;
-
-    SVGPathByteStream() { }
+    class Data final : public RefCounted<Data> {
+    public:
+        friend class NeverDestroyed<Data, MainThreadAccessTraits>;
+
+        using Bytes = Vector<uint8_t>;
+
+        static Ref<Data> create(Bytes&& bytes)
+        {
+            return adoptRef(*new Data(WTFMove(bytes)));
+        }
+
+        void updatePath(const Path& path) const
+        {
+            m_path = path;
+        }
+
+        static Ref<Data> empty()
+        {
+            static MainThreadNeverDestroyed<Data> singleton;
+            return Ref { singleton.get() };
+        }
+
+        const Bytes& bytes() const { return m_bytes; }
+
+        void append(uint8_t byte)
+        {
+            m_path = { };
+            m_bytes.append(byte);
+        }
+
+        void append(std::span<const uint8_t> bytes)
+        {
+            m_path = { };
+            m_bytes.append(bytes);
+        }
+
+        void append(const Bytes& other)
+        {
+            m_path = { };
+            m_bytes.appendVector(other);
+        }
+
+        const Path& path() const { return m_path; }
+
+        Ref<Data> copy() const
+        {
+            return adoptRef(*new Data(*this));
+        }
+
+        friend bool operator==(const Data& a, const Data& b)
+        {
+            return a.m_bytes == b.m_bytes;
+        }
+
+        unsigned size() const { return m_bytes.size(); }
+        bool isEmpty() const { return m_bytes.isEmpty(); }
+
+    private:
+        Data(Bytes&& bytes)
+            : m_bytes(WTFMove(bytes))
+        {
+        }
+
+        Data() = default;
+        Data(const Data& data)
+            : m_bytes(data.m_bytes)
+            , m_path(data.m_path)
+        {
+        }
+
+        Bytes m_bytes;
+        mutable Path m_path;
+    };
+
+    using DataIterator = Data::Bytes::const_iterator;
+
+    SVGPathByteStream()
+        : m_data(Data::empty())
+    {
+    }
 
     SVGPathByteStream(StringView string)
+        : SVGPathByteStream()
     {
         buildSVGPathByteStreamFromString(string, *this, UnalteredParsing);
     }
 
     SVGPathByteStream(const SVGPathByteStream& other)
-        : CanMakeSingleThreadWeakPtr<SVGPathByteStream>()
+        : SVGPathByteStream()
     {
         *this = other;
     }
 
     SVGPathByteStream(SVGPathByteStream&& other)
+        : SVGPathByteStream()
```

Comment:
Was it a mistake in the original code to have omit it?

Reply:
It is required now since `DataRef` does not have default constructor. So we need to initialize it explicitly.